src/rust/src/lib.rs

```diff
@@ -742,26 +749,124 @@ pub(crate) fn invoke_host_function(
     resources_buf: &CxxBuf,
     source_account_buf: &CxxBuf,
     auth_entries: &Vec<CxxBuf>,
-    ledger_info: CxxLedgerInfo,
+    mut ledger_info: CxxLedgerInfo,
     ledger_entries: &Vec<CxxBuf>,
     ttl_entries: &Vec<CxxBuf>,
     base_prng_seed: &CxxBuf,
     rent_fee_configuration: CxxRentFeeConfiguration,
 ) -> Result<InvokeHostFunctionOutput, Box<dyn std::error::Error>> {
     let hm = get_host_module_for_protocol(config_max_protocol, ledger_info.protocol_version)?;
-    (hm.invoke_host_function)(
+    let res = (hm.invoke_host_function)(
         enable_diagnostics,
         instruction_limit,
         hf_buf,
         resources_buf,
         source_account_buf,
         auth_entries,
-        ledger_info,
+        &ledger_info,
         ledger_entries,
         ttl_entries,
         base_prng_seed,
-        rent_fee_configuration,
-    )
+        &rent_fee_configuration,
+    );
+
+    // The rest of this is for divergence / comparison testing between protocols
+    if let Ok(extra) = std::env::var("SOROBAN_TEST_EXTRA_PROTOCOL") {
```

Comment:
Ideally, this block would be compiled out for release builds. What do you think about enabling a cargo feature on `BUILD_TESTS`, and gating this code on that feature?